docs/swagger.yaml

```diff
@@ -0,0 +1,381 @@
+        description: "Task object that needs to be added to the store"
+        required: true
+        schema:
+          $ref: "#/definitions/Task"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/Task"
+        "404":
+          description: "User not found"
+        "500":
+          description: "Internal server error"
+    
+  /users/{user_id}/tasks/{task_id}:
+    get:
+      tags:
+      - "users/{user_id}/tasks/{task_id}"
+      summary: "Get a user's task by task_id"
+      description: ""
+      operationId: "getTaskByName"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The id of the user who owns the task."
+        required: true
+        type: "integer"
+      - name: "task_id"
+        in: "path"
+        description: "The task id that needs to be fetched."
+        required: true
+        type: "integer"
+      responses:
+        "200":
+          description: "successful operation"
+          schema:
+            $ref: "#/definitions/User"
+        "400":
+          description: "Invalid username supplied"
+        "404":
+          description: "User not found"
+    put:
+      tags:
+      - "users/{user_id}/tasks/{task_id}"
+      summary: "Update a user's task with PUT method"
+      description: "This can only be done by the logged in user."
+      operationId: "updateTask"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The id of the user who owns the task."
+        required: true
+        type: "string"
+      - name: "task_id"
+        in: "path"
+        description: "The task id that will be updated."
+        required: true
+        type: "integer"
+      - in: "body"
+        name: "body"
+        description: "Updated user object"
+        required: true
+        schema:
+          $ref: "#/definitions/User"
+      responses:
+        "200":
+          description: "Successful operation"
+        "404":
+          description: "User not found"
+        "500":
+          description: "Internal server error"
+    patch:
+      tags:
+      - "users/{user_id}/tasks/{task_id}"
+      summary: "Update a user's task with PATCH method"
+      description: "This can only be done by the logged in user."
+      operationId: "patchTask"
+      produces:
+      - "application/json"
+      parameters:
+      - name: "user_id"
+        in: "path"
+        description: "The id of the user who owns the task."
+        required: true
+        type: "string"
+      - name: "task_id"
+        in: "path"
+        description: "The task id that will be updated."
+        required: true
+        type: "integer"
+      - in: "body"
+        name: "body"
+        description: "Updated user object"
+        required: true
+        schema:
+          $ref: "#/definitions/User"
```

Comment:
takes a Task object, and add a note that the fields are all optional - it will take whatever fields you pass in